lib/import.js

```diff
@@ -74,70 +73,104 @@ Import.prototype.url = function(url, options) {
 };
 
 
-
 Import.prototype._post = function(type, options) {
 
-  var self = this;
+  var self = this,
+    req,
+    reqOptions,
+    defaultReqOptions = {
+      url: self.import_api_url,
+      json: true,
+      qs: {
+        api_key: self.api_key,
+        privacy: options.privacy
+      }
+    };
 
+  reqOptions = options.reqOptions || defaultReqOptions;
+
+  debug(reqOptions)
   debug('POSTing your file to %s', self.import_api_url);
 
-  var req = request.post({
-    url: self.import_api_url,
-    json: true,
-    qs: {
-      api_key: self.api_key,
-      privacy: options.privacy
-    },
-  }, processResponse)
+  req = reqOptions.method ? request(reqOptions, processResponse)
+    : request.post(reqOptions, processResponse);
 
-  if(type == 'file') {
+  if(type === 'file') {
     req.form().append('file', fs.createReadStream(options.filePath));
   }
 
-  if(type == 'url') {
+  if(type === 'url') {
     req.form().append('url', options.url);
   }
 
-
   function processResponse(err, res, body) {
-     if (err) {
-       console.log(err);
-     } else {
-      console.log(body);
-      debug('Success, the item_queue_id is %s', body.item_queue_id)
+    if (err) {
+      options.promise.emit('_error', err);
+    } else {
+      body = requestUtil.fixBody(body);
+
+      debug(body);
+      debug('Success, the item_queue_id is %s', body.item_queue_id);
       if(body.item_queue_id) {
         poll(body.item_queue_id);
       }
-     }
+      else
+        options.promise.emit('_error', 'body.item_queue_id undefined');
+    }
   }
 
 
   function poll(queue_id) {
     debug('Polling to see if this file is finished importing...')
     setTimeout(function() {
-      var pollUrl = self.import_api_url + '/' + queue_id;
+    var pollUrl = self.import_api_url + '/' + queue_id;
 
-      request.get({
+      var getOptions = {
         url: pollUrl,
         json: true,
         qs: {
           api_key: self.api_key
         }
-      }, function(err, res, body) {
+      };
+
+      debug(getOptions);
+
+      request.get(getOptions, processGetResponsefunction);
 
-        if( body.state == 'complete' ) {
-          debug('It\'s finished! The table name is %s', body.table_name);
-          options.promise.emit('done', body.table_name)
 
-        } else if( body.state == 'failure') {
-          options.promise.emit('_error', body.get_error_text)
+      function processGetResponsefunction(err, res, body) {
 
-        } else {
-          debug('Not finished, trying again...')
-          poll(queue_id); //recurse
+        body = requestUtil.fixBody(body);
```

Comment:
This was one of the main problems in this library as body was almost never an object. This was probably due to not using proper versioning on `request` by setting it to `latest`.